src/index.ts

```diff
@@ -436,11 +451,20 @@ class QueryPipeline extends EventEmitter {
           }
         }
 
+        let promotionPinned = false;
+        if (activePins.length > 0) {
+          const val = valueFromProto(values["_id"]) as string;
+          if (activePins.indexOf(val) !== -1) {
```

Comment:
!!
This is the biggest thing we should check on. Do we actually need to check if the pin value, or can we simply use the pin's `position` value? If there is a pin with `position: 5`, I presume that we can just set `promotionPinned: true` on the 5th item in the results array.

Also note, the `position` value may be 1 based while the array is 0 based.

Reply:
That... makes a lot of sense 🙃